intelmq/lib/mixins/cache.py

```diff
@@ -51,6 +67,20 @@ def cache_set(self, key: str, value: Any, ttl: Optional[int] = None):
         if self.redis_cache_ttl:
             self.__redis.expire(key, self.redis_cache_ttl)
 
+    def cache_put(self, value: dict) -> int:
+        # Returns the length of the list after pushing
+        size = self.__redis.lpush(self.bot_id, json.dumps(value))
+        return size
```

Comment:
```suggestion
    def cache_lpush(self, key: str, value: Any) -> int:
        # Returns the length of the list after pushing
        size = self.__redis.lpush(key, value)
        return size
```

I appreciate the new function, but why is the name of the list not a parameter, but fixed?
Also, why is the value a dict?

The new functions are also missing in the docs in `docs/dev/bot-development.md`

Reply:
The idea was to create a message caching for the bot which would make it easy to use, completely abstract from the underlying implementation, not a simple wrapper over the Redis (see e.g. `cache_set`). This is why they do not follow exact the same naming as Redis library, and do additional things.

I have now mixed feelings in which direction it should go. I see two general options:
* renaming my new methods to something more clear, like `cache_dict`, `get_cached_dict`
* go with the simple wrapping over Redis library

I both cases, I would consider also making a key optional (and using bot ID as the default). If going with the first option, then I'd also handle transforming messages to the JSON and back. 

@sebix what do you think is better?

Reply:
> * renaming my new methods to something more clear, like `cache_dict`, `get_cached_dict`

These names do not indicate in any way that a FIFO queue is used, but instead make the impression that a dictionary is the cache.
What about `queue_push` and `queue_pop`?

>     * go with the simple wrapping over Redis library

It would be more generic and slightly lower-level. (However, redis/valkey are not that low-level anyway :) )

> I both cases, I would consider also making a key optional (and using bot ID as the default). If going with the first option, then I'd also handle transforming messages to the JSON and back.

I understand your desire for simplicity, but I have doubts about ambiguity. With an optional key, some possible interfaces are:
- `func(key: Optional[str], value: Any)`: Requires calling it with `None` as the first parameter: `func(None, foobar)`. No big difference to using `self.bot_id`: `func(self.bot_id, foobar)`
- `func(value: Any, key: Optional[str] = None)`: The order of parameters is the complete opposite of anything else in IntelMQ and elsewhere.
- `func(key_or_value: Any, value_if_key: Optional[Any])` allowing to call `func(value)` and `func(key, value)`: Too complicated. Not very KISSy :kiss:

Reply:
> I understand your desire for simplicity, but I have doubts about ambiguity. With an optional key, some possible interfaces are:

I'd never ever thought about option 3, and the option 1 is not simplifying anything :sweat_smile: So the 2 is the only one I would ever consider

But as my original idea seems to introduce too many questions, I'll move forward with your original idea of wrapping Redis library and let the developer use it flexible.